astropy/convolution/src/convolve.c

```diff
@@ -472,15 +518,22 @@ FORCE_INLINE void convolve3d_boundary_none(DTYPE * const result,
                         }}
                     }}
                 }}
+
+                if (padded) {
+                    result_index = (i_minus_wkx*ny_minus_2wky + j_minus_wky)*nz_minus_2wkz + k_minus_wkz;
+                } else {
+                    result_index = (i*ny + j)*nz + k;
+                }
+
                 if (nan_interpolate) // compile time constant
                 {
                     if (bot == 0) // This should prob be np.isclose(kernel_sum, 0, atol=normalization_zero_tol)
-                        result[(i*ny + j)*nz + k]  = f[(i*ny + j)*nz + k] ;
+                        result[result_index]  = f[(i*ny + j)*nz + k] ;
```

Comment:
This is probably an extremely tiny nit to pick, but there's another subexpression that could be factored out here in the source index.

Reply:
But this is even true for ``result_index``, i.e. hoist ``i*ny`` and ``j*nz``. I.e. it is not specific to this PR but to #7293

Reply:
Unless you meant something else?

Reply:
The compiler should figure this out, however, as with all the others, I had been explicit in making sure that it did by doing it myself. This is obviously something I missed (and the others similar).